syne_tune/blackbox_repository/conversion_scripts/scripts/tabrepo_import.py

```diff
@@ -0,0 +1,270 @@
+    "C": uniform(lower=0.1, upper=1e3),
+    "proc.skew_threshold": choice(["0.99", "0.9", "0.999", "None"]),
+    "proc.impute_strategy": choice(["median", "mean"]),
+    "penalty": choice(["L2", "L1"]),
+}
+
+CONFIGURATION_SPACE_RandomForest = {
+    "max_leaf_nodes": randint(5000, 50000),
+    "min_samples_leaf": choice([1, 2, 3, 4, 5, 10, 20, 40, 80]),
+    "max_features": choice(["sqrt", "log2", "0.5", "0.75", "1.0"]),
+}
+
+CONFIGURATION_SPACE_ExtraTrees = {
+    "max_leaf_nodes": randint(5000, 50000),
+    "min_samples_leaf": choice([1, 2, 3, 4, 5, 10, 20, 40, 80]),
+    "max_features": choice(["sqrt", "log2", "0.5", "0.75", "1.0"]),
+}
+
+CONFIGURATION_SPACE_NeuralNetTorch = {
+    "learning_rate": loguniform(1e-4, 3e-2),
+    "weight_decay": loguniform(1e-12, 0.1),
+    "dropout_prob": uniform(0.0, 0.4),
+    "use_batchnorm": choice([False, True]),
+    "num_layers": randint(1, 5),
+    "hidden_size": randint(8, 256),
+    "activation": choice(["relu", "elu"]),
+}
+
+CONFIGURATION_SPACE_XGBoost = {
+    "learning_rate": loguniform(lower=5e-3, upper=0.1),
+    "max_depth": randint(lower=4, upper=10),
+    "min_child_weight": uniform(0.5, 1.5),
+    "colsample_bytree": uniform(0.5, 1.0),
+    "enable_categorical": choice([True, False]),
+}
+
+CONFIGURATION_SPACE_LightGBM = {
+    "learning_rate": loguniform(
+        lower=5e-3,
+        upper=0.1,
+    ),
+    "feature_fraction": uniform(lower=0.4, upper=1.0),
+    "min_data_in_leaf": randint(lower=2, upper=60),
+    "num_leaves": randint(lower=16, upper=255),
+    "extra_trees": choice([False, True]),
+}
+
+CONFIGURATION_SPACE_CatBoost = {
+    "learning_rate": loguniform(lower=5e-3, upper=0.1),
+    "depth": uniform(lower=4, upper=8),
+    "l2_leaf_reg": uniform(
+        lower=1,
+        upper=5,
+    ),
+    "max_ctr_complexity": uniform(lower=1, upper=5),
+    "one_hot_max_size": choice([2, 3, 5, 10]),
+    "grow_policy": choice(["SymmetricTree", "Depthwise"]),
+}
+
+
+def generate_tabrepo(config_space, bb_name):
+    print(f"generating {bb_name}")
+
+    bb_dict = {}
+
+    # D244_F3_C1530 for full datasets
+    context_name = "D244_F3_C1530_3"
+
+    repo: EvaluationRepository = load_repository(
+        context_name, cache=True, load_predictions=False
+    )
+
+    metrics = repo.metrics(datasets=repo.datasets(), configs=repo.configs())
+    metrics = metrics[
+        metrics.index.get_level_values("framework").str.split("_").str[0]
+        == bb_name.split("_")[1]
+    ]
+    for dataset_name, group in metrics.groupby("dataset"):
+        print(f"Processing dataset: {dataset_name}")
+        hyperparameters_configurations = {}
+        for framework in group.index.to_frame(index=False)["framework"].values:
+            hyperparameters_configurations[framework] = repo.config_hyperparameters(
+                config=framework
+            )
+        bb_dict[dataset_name] = convert_dataset(
+            config_space, group, hyperparameters_configurations
+        )
```

Comment:
A comment would be useful to make the code easier to read (eg something like "We collect metrics for all frameworks and create for every dataset a tensor with shape ... that contains all evaluations.")